### DISCUSSION TITLE
How to make my github profile searchable on google?

### DISCUSSION THREAD

**Discussion by monouns** [community/community#22021] [New to GitHub]
<p>I made my github few years ago, and still not searchable on google…<br>
Both my profile and public master branch repositories…</p>
<p>How to make my github profile searchable on search engine?</p>
<p>Thanks in adnvance!</p>

**Comment by queenofcorgis** [comment]
<p>Hi <a class="mention" href="/monouns">@monouns</a> - welcome to the Community Forum :wave: <a href="https://github.community/t/github-profile-is-not-ranked-on-google-search/128855">This post</a> may help you out. Essentially keep being active in the community and contributing to open source!.</p>

**Comment by jfullstackdev** [comment, marked as answer]
<p>yup that link is correct, I experienced the same thing, I even installed Google Search Console and Google Analytics to speed it up,</p>
<p>the more links you are posting at every possible social networking sites, your traffic will increase, web crawlers will notice that and that’s it, you will be indexed</p>
<p>in short, decentralized links, be active in any forum like this, the more page you have, the greater the possibility and your username must be consistent, for me, I’m using the same username here and at Twitter, the web crawler will notice that too</p>

**Reply by janefudyma** [reply]
Can you send an example about how you used google search console to index your profile? I am having issues with it working. Thanks!

**Reply by farhan7reza7** [reply]
> Can you send an example about how you used google search console to index your profile? I am having issues with it working. Thanks!

Unless users having own website where they have full control over the codebase, GitHub users cannot use Google Search Console to index their GitHub account or profile because users do not have control over GitHub's website internal codes because Google Search Console requires verification through methods such as adding a meta tag to the <head> section of a website, which is not possible on GitHub, But if you have own website, then can index whatever urls or corresponding stuffs you want to index by linking the urls to your website called backlinking(which will index Github stuffs like profile or repository or etc.), same thing happen when share or post links or urls over websites....

**Reply by RakibDevHub** [reply]
I'm facing the same issue. After changing my GitHub username, I can't find my profile in Google Search results

**Reply by amanaanubhav** [reply]
> yup that link is correct, I experienced the same thing, I even installed Google Search Console and Google Analytics to speed it up,
> 
> the more links you are posting at every possible social networking sites, your traffic will increase, web crawlers will notice that and that’s it, you will be indexed
> 
> in short, decentralized links, be active in any forum like this, the more page you have, the greater the possibility and your username must be consistent, for me, I’m using the same username here and at Twitter, the web crawler will notice that too

I have listed my personal website onto the google search console and google analytics which has the links my github, linkedin etc. but still the google search can't configure and display my github while searching my name while my other profiles are being listed.

**Reply by HarshuP02** [reply]
my profile is not showing in both case before changing username or even after changing username

**Comment by mo-cherif** [comment]
keep contributing and blogging with the same username.

**Comment by TsvetanG2** [comment]
Thanks for the answer guys

**Comment by alessndro** [comment]
Hi there, the answer may come a little late, but what worked for me:

1. Put your repository on Public
2. I started following more people on Github, friends, co-workers and interesting people 
3. Put your GitHub link on different locations/ external links
4. Maintain and create a well structured profile, up to date information.
5. Get involved on forums and projects
6. Have patience

Cheers!

**Reply by 82GHOST** [reply]
ok bro cool down

**Reply by JosephScothmer-Clay** [reply]
Thanks! I'm on step 5 as you can probably tell by the comment.

**Reply by bluewater82** [reply]
> Thanks! I'm on step 5 as you can probably tell by the comment.

Me too! haha

**Reply by ciellansard** [reply]
> Thanks! I'm on step 5 as you can probably tell by the comment.

Me too, Joseph :-)

**Comment by niteesh18433** [comment]
Response #10240

**Comment by farhan7reza7** [comment]
> I made my github few years ago, and still not searchable on google… Both my profile and public master branch repositories…
> 
> How to make my github profile searchable on search engine?
> 
> Thanks in adnvance!

# Issue: 
Github by default not automatically index your profile by just username, but by using (Github Special repo) when search `@username/username` in search engine or  using the contents or headers of the special repo Readme or any other repositories contents or headers if any

# Simplest:
 Search your profile using `@username/username` in search engine after creating profile repo (Github Special repo) , then it should index because it is Github built in way for indexing your profile.

## Recommended:
 Create a Github special repo (if not already) with name same as your username `username` which is special repo name considered `Github profile repo`, and in README.md of this special repo(special understood by github), add  your profile data.

# For searching with just your username or name: 

in special repo Readme,  use specific format to write the profile details in order to include the keywords you want to use for searching in order to index your profile when you search using:

```markdown
# (first level header) - whatever keyword you want to search by consider keeping unique(phrase or word) if want at the top without doing anything extra for your account indexing(your name or username or anything) 

## (second level header) - title (keyword) for section of the profile 

Content for the section (paragraph) 



Add as many sections as you want for profile details
```

# For maintaining your profile indexing at top in search result:
 
- Keep you active and contributing to the respos, discussions, and hence keep adding contents about in your account which will be indexed as per above SEO

- link or post the urls of your github profile or repositories across other websites 

- if you have own website, then can index whatever urls or corresponding stuffs you want to index by linking the urls to your website using Google Search Console called backlinking(which will index Github stuffs like profile or repository or etc.), same thing happen when share or post links or urls over websites....

**Reply by dankunk** [reply]
Great advice. I'm going to reply to this post so hopefully it can boost me a bit more. Thanks Farhan!

**Reply by ForexPartnerCodes** [reply]
is this method works?

**Reply by wyy-47** [reply]
Just tried this... Hopefully it'll work, thanks!

**Reply by RicaBiz** [reply]
I'm gonna try this too. Hopefully it works!

**Reply by jimenezgimenojesus** [reply]
Doing the same here hahaha

**Reply by thecrackofdawn** [reply]
worth a try

**Comment by Anastasia22zx** [comment]
> <p>yup that link is correct, I experienced the same thing, I even installed Google Search Console and Google Analytics to speed it up,</p>
> <p>the more links you are posting at every possible social networking sites, your traffic will increase, web crawlers will notice that and that’s it, you will be indexed</p>
> <p>in short, decentralized links, be active in any forum like this, the more page you have, the greater the possibility and your username must be consistent, for me, I’m using the same username here and at Twitter, the web crawler will notice that too</p>

**Reply by csgocodes** [reply]
How you can check your githup profile with google serach console ??

**Comment by h4tem** [comment]
Hey, very helpful post thank you! 
I'm also trying to make my projects more discoverable, and I came across a simple trick that might help:

1 - Create a website (GitHub Pages is an easy option).
2 - Include a reference to your GitHub repository on that site.
3 - Use Google Search Console to index your webpage and improve its visibility on Google.

I just set up a quick example here: https://h4tem.github.io/ – I’ll check back in a few days to update on whether this method worked!

**Reply by Sriram-PR** [reply]
did it work?

**Reply by h4tem** [reply]
Hey,
Yeah, my GitHub profile is now the first result when you search my username on Google!

However, I don't think setting up the GitHub Page alone made it happen. I believe that being active, i.e. like opening issues and interacting with others (just like we're doing here) played a bigger role :)

**Reply by RicaBiz** [reply]
I already have a github pages website, so I'll try this xD let's hope

**Comment by ElliotXie** [comment]
It has been two years, and my profile is still not searchable.

**Reply by aflacake** [reply]
When searching use the word "github <YOUR_USERNAME>"

**Reply by thelanceprogramming** [reply]
I've tried this "github <MY_USERNAME>" it still doesn't work :( bummer

**Comment by RicaBiz** [comment]
As suggested by both [farhan7reza7](https://github.com/farhan7reza7) and [h4tem](https://github.com/h4tem) I made a repository and I added my github Pages Website into Google Search Console, as well as making this very post.
I will report here in a couple days to check if it worked.

**Reply by RicaBiz** [reply]
One day passed. As of right now, my Pages website appears without a problem on google sesrch, but still no sign of my GitHub profile or public repositories. 👍

**Comment by JoeBumm** [comment]
GitHub profiles and repositories are not immediately indexed by search engines. To make them searchable:

Set your profile and repos to public (sounds like you’ve done this already).

Add a meaningful bio and repository descriptions—search engines index these.

Add a README.md to your repos with clear content and keywords.

Link to your GitHub profile or repos from other websites, blogs, or social media—Google follows links to index pages faster.

Be patient: It can take days to weeks for Google to crawl and index new or updated pages.

Submit your profile/repo URLs to Google Search Console
 to speed up indexing.

**Comment by thelanceprogramming** [comment]
The same for me! it's been 2 years since my account and I already have a bunch of public repositories yet it is still not being indexed!

**Comment by yesanqian** [comment]
In fact, about a month ago, I searched for relevant information on this topic.

I've created a GitHub repository for those who want to bypass Telegram channel or group copy-forward restrictions (e.g., restricted saving/downloading of content)(Restrict saving content,Telegram  download restricions):

https://github.com/yesanqian/tg-message-clone

As you can see, it has now been successfully indexed by Google.

**First, I want to sincerely thank everyone who offered advice on this matter!**

Here are the steps I took to help with the indexing:

> 1. I updated my profile and uploaded a new avatar.
> 
> 2. I shared the repository link on other Google-indexed sites (for better visibility and backlinks, which may take time to take effect). 
> 
> 3. I participated in some interactions on GitHub, such as joining discussions.
> 
> 4. I forked someone else's project.
> 
> 5. I waited for about a month.

**Hopefully, these steps will help others who are trying to solve the same problem.**

**Comment by IPTVMap** [comment]
**I have the same problem too**

**Comment by GLAXYIAS** [comment]
well, I am a low-key user and my stuff got indexed quickly and I didn't do Google search console, shared nothing, just have 24 followers 2 good repos, each have three stars, and have over 1,400 commits, and I got indexed, the only thing I can think of is I just added robots.txt and sitemap.xml
